### ISSUE TITLE
Question about thread mode

### ISSUE THREAD

**Issue by prsolucoes** [SRombauts/SQLiteCpp#195]
Hi,

We have some problems using your lib with conan because the conan sqlitecpp receipt has option "threadsafe=2" that set sqlite3 to be compiled using ```SQLITE_THREADSAFE=2```.

But we always use your library (sqlitecpp) with ```SQLITE_THREADSAFE=1``` and don't have problems.

Can you explain why i can't use ```SQLITE_THREADSAFE=1``` with your library? 

If i enable ```SQLITE_THREADSAFE=2``` on android and ios i got a lot of problems and when run ios app that app crashed.

Can you help me understand?

Thanks.

**Comment by SRombauts** [comment]
Hi @prsolucoes, I am sorry my answer never reached you; I wanted to wrote here that my SQLiteCpp wrapper does not really care of what SQLITE_THREADSAFE value you compile it with; SQLITE_THREADSAFE=1 should work for most use case

**Comment by prsolucoes** [comment]
Ok.

So my suggestion is that you remove in your readme the text that you don't support other sqlite thread safe.

The conan repository is forcing the sqlite be compile with multithread because in readme you say that don´t support other methods.

Can you do this?

Thanks.

**Comment by SRombauts** [comment]
You are right, I need to review the README: I think I re-wrote it after similar in the past, but I should be more explicit; the serialized option would work, but you would still have to take care of not sharing statements between threads due to the fact that the wrapper is not thread safe itself

**Comment by Al2Klimov** [comment]
Hello @SRombauts!

It the way you access `SQLite::Statement::Ptr#mpRefCount` the only reason why your lib isn't fully thread-safe?

If yes, shall I make it a `std::atomic<unsigned int>*`?

Best,
AK

**Comment by paulocoutinhox** [comment]
Any update?
